File: NavigationSampleSafeJava/app/src/main/java/com/websarva/wings/android/navigationsamplesafejava/data/repository/MemoRepository.java
package com.websarva.wings.android.navigationsamplesafejava.data.repository;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.websarva.wings.android.navigationsamplesafejava.data.model.Memo;

import java.util.ArrayList;
import java.util.List;

public class MemoRepository {
	private final List<Memo> _memoList = new ArrayList<>();

	public MemoRepository() {
		for(int i = 1; i <= 30; i++) {
			Memo memo = new Memo();
			memo.setId((long) i);
			memo.setTitle("メモ" + i + "のタイトル");
			memo.setContent("これはメモ内容" + i + "です。");
			_memoList.add(memo);
		}
	}

	public LiveData<List<Memo>> prepareMemoListLiveData() {
		MutableLiveData<List<Memo>> memoListLiveData = new MutableLiveData<>();
		memoListLiveData.setValue(_memoList);
		return memoListLiveData;
	}

	public Memo getOneMemo(long id) {
		return _memoList.stream().filter(memo -> memo.getId() == id).findFirst().orElse(null);
	}
}

File: NavigationSampleSafeKotlin/app/src/main/java/com/websarva/wings/android/navigationsamplesafekotlin/data/repository/MemoRepository.kt
package com.websarva.wings.android.navigationsamplesafekotlin.data.repository

import com.websarva.wings.android.navigationsamplesafekotlin.data.model.Memo
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.MutableStateFlow

class MemoRepository {
	private val _memoList: List<Memo>

	init {
		_memoList = mutableListOf()
		for(i in 1..30) {
			val memo = Memo(i.toLong(), "メモ${i}のタイトル", "これはメモ内容${i}です。")
			_memoList.add(memo)
		}
	}

	fun prepareMemoListFlow(): Flow<List<Memo>> {
		return MutableStateFlow(_memoList)
	}

	fun getOneMemo(id: Long): Memo? {
		return _memoList.stream().filter { memo -> memo.id == id }.findFirst().orElse(null)
	}
}
